src/files/manager.ts

```diff
@@ -229,4 +237,117 @@ export class FilesManager {
       throw error;
     }
   }
+
+  /**
+   * Uploads a new media file to the Strapi Media Library.
+   *
+   * @param file - The file to upload (Blob or Buffer).
+   * @param options - Upload options. For Buffer uploads, filename and mimetype are required.
+   * @returns A promise that resolves to the uploaded file information.
+   *
+   * @throws {FileForbiddenError} if the user does not have permission to upload files.
+   * @throws {HTTPError} if the HTTP client encounters connection issues, the server is unreachable, or authentication fails.
+   *
+   * @example
+   * ```typescript
+   * const filesManager = new FilesManager(httpClient);
+   *
+   * // Upload with a File object (browser)
+   * const fileInput = document.querySelector('input[type="file"]');
+   * const file = fileInput.files[0];
+   * const result = await filesManager.upload(file, {
+   *   fileInfo: { alternativeText: 'A user uploaded image' }
+   * });
+   *
+   * // Upload with a Blob
+   * import { blobFrom } from 'node-fetch';
+   * const file = await blobFrom('./1.png', 'image/png');
+   * const result = await filesManager.upload(file, {
+   *   fileInfo: { alternativeText: 'An example image' }
+   * });
+   *
+   * // Upload with a Buffer (Node.js) - filename and mimetype are required
+   * import fs from 'fs';
+   * const buffer = fs.readFileSync('./image.png');
+   * const result = await filesManager.upload(buffer, {
+   *   filename: 'image.png',
+   *   mimetype: 'image/png',
+   *   fileInfo: { alternativeText: 'An example image' }
+   * });
+   *
+   * console.log(result); // Upload response with file details
+   * ```
+   */
+  async upload(file: Blob, options?: BlobUploadOptions): Promise<MediaUploadResponse>;
+  async upload(file: Buffer, options: BufferUploadOptions): Promise<MediaUploadResponse>;
+  async upload(
+    file: Blob | Buffer,
+    options?: BlobUploadOptions | BufferUploadOptions
+  ): Promise<MediaUploadResponse> {
```

Comment:
@Bassel17 and @maccomaccomaccomacco 

Could you add another voice to this method signature in particular? Jamie and I think we've got it, but since changing this would be a breaking change later I want to make sure we're not missing something.

Usage would be one of those three options written in the header comment

Reply:
Looks fine, even if we add new stuff to it later it will just be additions and won't be breaking changes, like uploading maybe multiple files or something like that